client/core/trade.go

```diff
@@ -1541,28 +1537,73 @@ func (t *trackedTrade) processRedemption(msgID uint64, redemption *msgjson.Redem
 	return nil
 }
 
-// hasUnswappedMatches will be true if the order has any remaining matches for
-// which our swap has not been broadcast. hasUnswappedMatches should be called
-// with the matchMtx locked.
-func (t *trackedTrade) hasUnswappedMatches() bool {
+// Coins will be returned if
+// - the trade status is not OrderStatusEpoch or OrderStatusBooked, that is to
+//   say, there won't be future matches for this order.
+// - there are no matches in the trade that MAY later require sending swaps,
+//   that is to say, all matches have been either swapped or revoked.
+//
+// This method modifies match fields and MUST be called with the trackedTrade
+// mutex lock held for writes.
+func (t *trackedTrade) maybeReturnCoins() bool {
```

Comment:
What do we think about changing the `ReturnCoins` to `UnlockCoins` consistently. Return coins sounds as though coins were moved from the wallet but should now be returned to the wallet. Unlock sends the message that the coins were locked (to prevent unplanned spending) and should now be unlocked to permit spending anyhow.

Reply:
Err, I see. `UnlockCoins` is sort of implementation-biased. We do have a `Locked` balance in dexc (implementation-agnostic) and there are a couple other references to locking coins such as in the doc for `FundOrder`.

Probably not a big deal / head turner but sticking with Lock/Unlock wouldn't be necessarily be a reflection of a utxo-based blockchain. It'd just be harmonising expressions already used elsewhere in the codebase.